### PR TITLE
Add woocommerce attributes to index data

### PR DESCRIPTION
**Pull request by SublimeProphets**
Hi there,

I like your plugin, fits the basic need very well.
To be able to create some advanced faceting filters I had the need to add the attributes to the index.
May you are interested into this small code addition. 
I kept it straightforward and as simple as possible.

Hope you like it and you may can use.
Some further improvements here would be to add settings for the backend so the user could toggle the indexing of data:
would be nice if the submitted data could be selected - the attributes could be done by listing all available attributes from woocommerce and checkboxes :-) I am quiet short in time so i just added it straight to the code like all the other data as well.

If you have some questions, feel free to ask.

Greets Chris

### CODE REVIEW

**Comment by w3bdesign** [comment]
@SublimeProphets 

Thanks for the contribution, but I am getting some errors from Codacy, can you take a look?

**Comment by SublimeProphets** [comment]
> @SublimeProphets
> 
> Thanks for the contribution, but I am getting some errors from Codacy, can you take a look?

Hi @w3bdesign 
sorry i messed up the PR, instead of my featurebranch I selected the master branch which contains some for my use case specific code.
Nevertheless this motivates me to make a better implementation which does include some more stuff (categories, tags, metadata and more. 

Until this is finished, can you provide me some information about how i can check codacy locally? 

Greets

**Comment by w3bdesign** [comment]
@SublimeProphets 

You can create a fork and setup Codacy on your application.

If you need help, let me know

File: classes/class-send-products.php [hunk 1]
```diff
@@ -105,6 +104,71 @@ public static function get_product_type_price($product)
...
+         * Checks if stock management is enabled and if so, returns quantity and status
+         *
+         * @param  mixed $product Product to check   
+         * @return array ['stock_quantity' => $stock_quantity,'stock_status' => $stock_status] Array with quantity and status. if stock management is disabled, false will be returned,
+         */
+        public static function get_product_stock_data($product)
+        {
+            if ($product->get_manage_stock()) {
+                return array(
+                    'stock_quantity' => $product->get_stock_quantity(),
```
Comment (w3bdesign) [inline classes/class-send-products.php:118]: @SublimeProphets 

Does this handle all type of products, like variable products and simple products?

File: classes/class-send-products.php [hunk 2]
```diff
@@ -105,6 +104,71 @@ public static function get_product_type_price($product)
...
+        /**
+         * Get attributes from product
+         *
+         * @param  mixed $product Product to check   
+         * @return array ['pa_name' => ['value1', 'value2']] Array with key set to the product attribute internal name and values as array. returns false if not attributes found.
+         */
+        public static function get_product_attributes($product)
+        {
+            $rawAttributes = $product->get_attributes();
+            $numericRangeAttributes = ["pa_height", "pa_flowermonth"];
```
Comment (w3bdesign) [inline classes/class-send-products.php:135]: @SublimeProphets 

Why are the table names hardcoded?

**Review by w3bdesign — CHANGES_REQUESTED** [review]
@SublimeProphets 

See my comments.

File: classes/class-send-products.php (hunk 1 above)
Comment (SublimeProphets) [reply classes/class-send-products.php:118]: yes, it works for both variable and simple products. it does always retrieve the data from the main tab, the data for variable products is not used.

File: classes/class-send-products.php (hunk 2 above)
Comment (SublimeProphets) [reply classes/class-send-products.php:135]: As mentioned above this was not intended. I will close this PR and open a new one with plenty of added features :)

**Comment by SublimeProphets** [comment]
closed in favor on https://github.com/w3bdesign/algolia-woo-indexer/pull/169

**Pull request closed without merging**